src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,117 +1,227 @@
+  React.InputHTMLAttributes<HTMLInputElement>,
+  'checked' | 'id' | 'size'
+>;
+
+type InputProps = CheckboxHTMLElementProps & {
   /**
-   * Toggles whether or not the checkbox is checked or unchecked
+   * Whether checkbox is checked. Defaults to false.
+   * "indeterminate" can be used when a checkbox visually represents
+   * a list of checkboxes that is "partially" checked.
    */
-  checked?: boolean;
+  checked?: boolean | 'indeterminate';
   /**
-   * CSS class names that can be appended to the component.
+   * Additional classnames passed in for styling.
    */
   className?: string;
   /**
-   * Disables the field and prevents editing the contents
+   * Checkbox ID. Used to connect the input with a label for accessibility purposes.
    */
-  disabled?: boolean;
+  id: string;
+};
+
+type LabelProps = {
   /**
-   * HTML id for the component
+   * Text to render in label.
    */
-  id?: string;
+  text: React.ReactNode;
   /**
-   * Indeterminate prop that turns the checkbox indeterminate state on
+   * Additional classnames passed in for styling.
    */
-  indeterminate?: boolean;
+  className?: string;
   /**
-   * Inverted variation for dark backgrounds
+   * ID of input that label is associated with.
    */
-  inverted?: boolean;
+  htmlFor: string;
   /**
-   * HTML name attribute for the checkbox
+   * Size of the checkbox label.
    */
-  name?: string;
+  size?: 'small' | 'medium';
+};
+
+// id is required in InputProps but optional in CheckboxProps, so we
+// first remove `id` from InputProps before intersecting.
+export type CheckboxProps = Omit<InputProps, 'id'> & {
   /**
-   * Function that fires when field value has changed
+   * HTML id attribute. If not passed, this component
+   * will generate an id to use for accessibility.
    */
-  onChange?: ChangeEventHandler;
+  id?: string;
   /**
-   * Toggles the form control's interactivity. When `readOnly` is set to `true`, the form control is not interactive
+   * Visible text label for the checkbox.
    */
-  readOnly?: boolean;
+  label?: React.ReactNode;
   /**
-   * A string representing the value of the checkbox
+   * Size of the checkbox label.
    */
-  value?: string;
+  size?: LabelProps['size'];
+};
+
+interface SvgStyle extends React.CSSProperties {
+  '--checkbox-svg-size': string;
 }
 
 /**
- * Primary UI component for user interaction
+ * Variables controlling checkbox SVG size.
+ * Surfaced here since they're used in the input, label, and CSS
  */
-export const Checkbox = ({
-  ariaDescribedBy,
-  id,
-  name,
-  value,
-  className,
-  inverted,
-  indeterminate = false,
-  checked,
-  disabled,
-  readOnly,
-  onChange,
-  ...other
-}: Props) => {
-  const checkboxRef = useRef() as MutableRefObject<HTMLInputElement>;
-  /**
-   * On Keydown
-   */
-  function onKeyDown(e: any) {
-    /**
-     * If enter key is pressed, trigger on change event
-     */
-    if (onChange && e.code === ENTER_KEYCODE) {
-      onChange(e);
-    }
-  }
```

Comment:
There used to be an 'enter' key listener but I never got it to work, so just removed it


https://user-images.githubusercontent.com/86632227/162302708-b5b82f9d-c501-4d7f-ba5e-f14ed9c89097.mov